src/core/annotation.js

```diff
@@ -59,25 +59,39 @@ class AnnotationFactory {
    * @param {Object} ref
    * @param {PDFManager} pdfManager
    * @param {Object} idFactory
+   * @param {boolean} checkJSActions
    * @returns {Promise} A promise that is resolved with an {Annotation}
    *   instance.
    */
-  static create(xref, ref, pdfManager, idFactory) {
-    return pdfManager.ensureCatalog("acroForm").then(acroForm => {
+  static create(xref, ref, pdfManager, idFactory, checkJSActions = true) {
+    return Promise.all([
+      pdfManager.ensureCatalog("acroForm"),
+      checkJSActions
+        ? pdfManager.ensureDoc("hasJSActions")
+        : Promise.resolve(false),
```

Comment:
I believe that the following ought to suffice, since non-Promise values should be automatically wrapped in a Promise:
```suggestion
      checkJSActions && pdfManager.ensureDoc("hasJSActions"),
```